homeassistant/components/venstar/sensor.py

```diff
@@ -78,58 +120,61 @@ def unique_id(self):
         """Return the unique id."""
         return f"{self._config.entry_id}_{self.sensor_name.replace(' ', '_')}_{self.entity_description.key}"
 
-
-class VenstarHumiditySensor(VenstarSensor):
-    """Represent a Venstar humidity sensor."""
-
     @property
     def name(self):
         """Return the name of the device."""
-        return f"{self._client.name} {self.sensor_name} Humidity"
+        return self.entity_description.name_fn(self.coordinator, self.sensor_name)
 
     @property
     def native_value(self) -> int:
         """Return state of the sensor."""
-        return self._client.get_sensor(self.sensor_name, "hum")
-
-
-class VenstarTemperatureSensor(VenstarSensor):
-    """Represent a Venstar temperature sensor."""
-
-    @property
-    def name(self):
-        """Return the name of the device."""
-        return (
-            f"{self._client.name} {self.sensor_name.replace(' Temp', '')} Temperature"
-        )
+        return self.entity_description.value_fn(self.coordinator, self.sensor_name)
 
     @property
     def native_unit_of_measurement(self) -> str:
         """Return unit of measurement the value is expressed in."""
-        if self._client.tempunits == self._client.TEMPUNITS_F:
-            return TEMP_FAHRENHEIT
-        return TEMP_CELSIUS
-
-    @property
-    def native_value(self) -> float:
-        """Return state of the sensor."""
-        return round(float(self._client.get_sensor(self.sensor_name, "temp")), 1)
+        return self.entity_description.uom_fn(self.coordinator)
 
 
 SENSOR_ENTITIES: tuple[VenstarSensorEntityDescription, ...] = (
     VenstarSensorEntityDescription(
-        key="humidity",
+        key="hum",
         device_class=DEVICE_CLASS_HUMIDITY,
         state_class=STATE_CLASS_MEASUREMENT,
-        native_unit_of_measurement=PERCENTAGE,
-        cls=VenstarHumiditySensor,
-        stype="hum",
+        uom_fn=lambda coordinator: PERCENTAGE,
+        value_fn=lambda coordinator, sensor_name: coordinator.client.get_sensor(
+            sensor_name, "hum"
+        ),
+        name_fn=lambda coordinator, sensor_name: f"{coordinator.client.name} {sensor_name} Humidity",
     ),
     VenstarSensorEntityDescription(
-        key="temperature",
+        key="temp",
         device_class=DEVICE_CLASS_TEMPERATURE,
         state_class=STATE_CLASS_MEASUREMENT,
-        cls=VenstarTemperatureSensor,
-        stype="temp",
+        uom_fn=lambda coordinator: TEMP_FAHRENHEIT
```

Comment:
Since the lambda spans more than one line it's easier to read if we just define a normal function above at the module level.
```py
def temperature_unit(coordinator: VenstarDataUpdateCoordinator) -> str:
    """Return the correct unit for temperature."""
    unit = TEMP_CELSIUS
    if coordinator.client.tempunits == coordinator.client.TEMPUNITS_F:
        unit = TEMP_FAHRENHEIT
    return unit

...

VenstarSensorEntityDescription(
    ...
    uom_fn=temperature_unit,
    ...
```